pkg/apis/v1alpha1/nsxserviceaccount_types.go

```diff
@@ -0,0 +1,86 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NSXServiceAccountSpec defines the desired state of NSXServiceAccount
+type NSXServiceAccountSpec struct {
+	VPCName string `json:"vpcName,omitempty"`
+}
+
+type NSXProxyEndpointAddress struct {
+	Hostname string `json:"hostname,omitempty"`
+	//+kubebuilder:validation:Format=ip
+	IP string `json:"ip,omitempty"`
+}
+
+type NSXProxyProtocol string
+
+const (
+	NSXProxyProtocolTCP NSXProxyProtocol = "TCP"
+)
+
+type NSXProxyEndpointPort struct {
+	Name     string           `json:"name,omitempty"`
+	Port     uint16           `json:"port,omitempty"`
+	Protocol NSXProxyProtocol `json:"protocol,omitempty"`
+}
+
+type NSXProxyEndpoint struct {
+	Addresses []NSXProxyEndpointAddress `json:"addresses,omitempty"`
+	Ports     []NSXProxyEndpointPort    `json:"ports,omitempty"`
+}
+
+type NSXSecret struct {
```

Comment:
Why redeclare this type when [`corev1.SecretReference`](https://pkg.go.dev/k8s.io/api/core/v1#SecretReference) exists?